submissions/chimoney-telegram-bot/src/start.js

```diff
@@ -0,0 +1,191 @@
+const { Telegraf, Markup } = require('telegraf');
+const axios = require("axios")
+const fs = require("fs")
+const express = require('express');
+
+
+const PORT = 3066;
+const token = "";
+const bot_url = "https://t.me/chi_amosayomide05_tg_bot"
+const api_key = ""
+const redirect_url = ""
+
+
+const bot = new Telegraf(token);
+const app = express();
+
+
+
+app.get('/confirm', (req, res) => {
+    const paymentRef = req.query.issueID;
+    const status = req.query.status;
+
+    if (status === "success") {
+        const readPayFile = JSON.parse(fs.readFileSync("payments.json", "utf8"));
+        var findPay = readPayFile.find(item => item.paymentRef === paymentRef);
+
+        if (findPay) {
+
+            var keyboard = Markup.inlineKeyboard([
+                Markup.button.url("Message Me", bot_url + "?start=claim_" + paymentRef)
+            ])
+            var chatId = findPay.chatId;
+            var receiverName = findPay.receiverName;
+            var senderName = findPay.senderName;
+
+            bot.telegram.sendMessage(chatId, `@${receiverName} You have received chimoney from @${senderName}, \n\nClick the below link to claim `, keyboard)
+        }
+    }
+    res.redirect(bot_url);
+});
+
+app.listen(PORT, () => {
+    console.log("http://localhost:3066")
+    console.log(`Server is running on port ${PORT}`);
+});
+
+
+
+bot.start((ctx) => {
+
+    const checkUsers = JSON.parse(fs.readFileSync("users.json", "utf8"));
+    var searchUser = checkUsers.find(item => item.firstName === ctx.message.from.first_name);
+
+    if (!searchUser) {
+        var user_details = {}
+        user_details.firstName = ctx.message.from.first_name;
+        user_details.id = ctx.message.from.id;
+        user_details.username = ctx.message.from.username;
+
+        checkUsers.push(user_details);
+        fs.writeFileSync("users.json", JSON.stringify(checkUsers, null, 2));
+
+    }
+
+    if (ctx.message.text.includes("claim_")) {
+        var msg = ctx.message.text;
+
+        var paymentRef = msg.slice(13)
+
+        const readPayFile = JSON.parse(fs.readFileSync("payments.json", "utf8"));
+
+
+        var findPay = readPayFile.find(item => (item.paymentRef === paymentRef) && (item.receiverName === ctx.message.from.username));
+
+        if (findPay) {
+
+            var redeemLink = "https://dash.chimoney.io/redeem?chiRef=" + findPay.chiRef
+
+            var keyboard = Markup.inlineKeyboard([
+                Markup.button.url("Redeem", redeemLink)
+            ])
+
+            var msg = `Congrats!!!, You've received $${findPay.amount} from @${findPay.senderName}.\n\nRedeem Now:`;
+            ctx.reply(msg, keyboard)
+        }
+    }
+
+
+});
+
+
+bot.on('message', async (ctx) => {
+
+    if (ctx.message.chat.type === "group" || ctx.message.chat.type === "supergroup") {
+        if (ctx.message.text.toLowerCase().startsWith('/pay ')) {
```

Comment:
Thoughts on using a more specific slash command because `/pay` might be another both command? Consider `/chimoney_pay`